packages/api-request-builder/src/create-service.js

```diff
@@ -42,7 +44,12 @@ export default function createService (
   if (!Array.isArray(definition.features) || !definition.features.length)
     throw new Error('Definition requires `features` to be a non empty array.')
 
+  if (!project.projectKey)
+    throw new Error('No project defined. Please enter a project key')
```

Comment:
You are validating twice (here and in `create-request-builder.js`). Just pick one ;)